pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -305,19 +305,20 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 		}
 
 		if managementCluster != nil {
-			tunnelCASecret, err = certificateManager.GetKeyPair(r.client, render.VoltronTunnelSecretName, common.OperatorNamespace())
-			if tunnelCASecret == nil {
-				tunnelSecret, err := certificatemanagement.CreateSelfSignedSecret(render.VoltronTunnelSecretName, common.OperatorNamespace(), "tigera-voltron", []string{"voltron"})
-				if err == nil {
-					tunnelCASecret = certificatemanagement.NewKeyPair(tunnelSecret, nil, "")
-					// Creating the voltron tunnel secret is not (yet) supported by certificate mananger.
-					tunnelSecretPassthrough = render.NewPassthrough(tunnelCASecret.Secret(common.OperatorNamespace()))
-				}
-			}
+			tunnelCASecret, err := utils.GetSecret(ctx, r.client, render.VoltronTunnelSecretName, common.OperatorNamespace())
 			if err != nil {
-				r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to get or create the tunnel secret", err, reqLogger)
+				r.status.SetDegraded(operatorv1.ResourceReadError, "Unable to fetch the tunnel secret", err, reqLogger)
 				return reconcile.Result{}, err
 			}
+			if tunnelCASecret == nil {
+				tunnelCASecret, err = certificatemanagement.CreateSelfSignedSecret(render.VoltronTunnelSecretName, common.OperatorNamespace(), "tigera-voltron", []string{"voltron"})
```

Comment:
Something seems funny about the fact that nothing was broken with MCM without the operator creating this secret. If the operator was supposed to be responsible for creating this, wouldn't every new MCM cluster be broken on startup?

Are you able to point to where the regression happen, and the code to create this was removed?

Reply:
First of all, the issue only occurs when installation.spec.certificatemanagement is enabled. This feature is probably only ever tested by me and 1 other person doing test cycles with both MCM and installation.spec.certificatemanagement. On top of that if you first create the management cluster and then enable certificate management, the secret will have already been created.

Looking back, I think the issue was present when we first merged b0fce5783d50c41c30942494fa1977b2d2d9325d

Reply:
If my memory serves me well during code reviews and testing, this line has changed: https://github.com/tigera/operator/commit/b0fce5783d50c41c30942494fa1977b2d2d9325d#diff-f9dc3b7ddca285064281a90259a65ddf0b5949a3ea1ba03ada3bda3e12810a0fR190

Reply:
> On top of that if you first create the management cluster and then enable certificate management, the secret will have already been created.

I'm still a little confused about what exactly is going on here. What's creating the secret when certificate management is not enabled? Is that in the operator?

Reply:
```
tunnelSecretPassthrough = render.NewPassthrough(tunnelCASecret)
...
	if tunnelSecretPassthrough != nil {
		components = append(components, tunnelSecretPassthrough)
	}
```
It is added by rendering a passthrough component.

Reply:
I think I had a brain fart when I first asked these questions, I thought previous there was no `certificatemanagement.CreateSelfSignedSecret` call before your changes if `certificateManager.GetKeyPair` return an nil, so I was wondering what the heck used to create this. I had just missed the glaringly obvious section that you just moved.

You've only renamed the variable and handled the errors properly.

Forgot _**all**_ of this...